File: packages/backend/src/main/kotlin/quizzem/backend/features/game/api/dto/GameDto.kt
package quizzem.backend.features.game.api.dto

import quizzem.backend.core.api.dto.QuizzemDto
import quizzem.backend.features.game.model.GameState
import quizzem.backend.features.question.api.dto.QuestionDto
import java.time.Instant
import java.util.*

class GameDto(
    id: UUID,
    createdAt: Instant,
    updatedAt: Instant,
    val state: GameState,
    val title: String?,
    val description: String?,
    val questions: List<QuestionDto>
) : QuizzemDto(id, createdAt, updatedAt)

File: packages/backend/src/features/game/api/dto/game.dto.ts
import { QuizzemDto } from 'src/core/api/quizzem.dto';
import { GameState } from 'src/features/game/model/game-state.model';
import { QuestionDto } from 'src/features/question/api/dto/question.dto';

export class GameDto extends QuizzemDto {
  state!: GameState;
  title!: string | null;
  description!: string | null;
  questions!: QuestionDto[];
}
